File: GrooveOn.Models/RequestObjects/AnswerUpsertRequest.cs
using System.ComponentModel.DataAnnotations;

namespace GrooveOn.Model.RequestObjects
{
    public class AnswerUpsertRequest
    {
        [Required]
        public int QuestionId { get; set; }

        [Required]
        public int AdminId { get; set; }

        [Required]
        [MinLength(2)]
        public string Message { get; set; } = string.Empty;
    }
}

File: UI/grooveon_desktop/lib/models/request/answer_upsert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'answer_upsert_request.g.dart';

@JsonSerializable()
class AnswerUpsertRequest {
  final int questionId;
  final int adminId;
  final String message;

  AnswerUpsertRequest({
    required this.questionId,
    required this.adminId,
    required this.message,
  });

  factory AnswerUpsertRequest.fromJson(Map<String, dynamic> json) =>
      _$AnswerUpsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$AnswerUpsertRequestToJson(this);
}